airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py

```diff
@@ -84,6 +84,25 @@ def read_records_from_slice(self, stream_slice: StreamSlice) -> Iterable[Mapping
             try:
                 for record in parser.parse_records(self.config, file, self._stream_reader, self.logger):
                     line_no += 1
+                    if record is None:
```

Comment:
I have some hesitations about letting `parse_records` return `None`. If it's indicative of an invalid file, I feel like it would be more intuitive to raise an exception and stop trying to parse the file, and think that would more closely mirror what the existing S3 connector does. WDYT?